Sources/MapboxMaps/Offline/OfflineErrors.swift

```diff
@@ -0,0 +1,110 @@
+import Foundation
+
+internal protocol CoreErrorRepresentable {
+    associatedtype CoreErrorType
+    init(coreError: CoreErrorType)
+    init(unspecifiedError: String)
+}
+
+/// Describes the reason for a tile region download request failure.
+public enum TileRegionError: LocalizedError, CoreErrorRepresentable {
```

Comment:
This feels similar to the extensible enum pattern due to the `other` case: https://github.com/mapbox/mapbox-maps-ios/pull/285